### PR TITLE
MF-296 - Create docker-compose with InfluxDB and Grafana

### LINKED ISSUE #296: Create docker-compose with InfluxDB and Grafana
Create new `docker-compose-influxdb-grafana.yaml` that includs well - InfluxDB and Grafana.

This should be done to facilitate demos (so easy to deploy and well documented).

Full example of usage should be done in the form of blog post on official Mainflux Medium blog, including screenshots and detailed instructions how to set-up Grafana and get some graphs from messages sent via MQTT from devices.

### PR DESCRIPTION
**Pull request by dborovcanin**
This pull request resolves #296.

Signed-off-by: Dušan Borovčanin <dusan.borovcanin@mainflux.com>

### CODE REVIEW

**Comment by nmarcetic** [comment]
@dusanb94  Do you think its better to keep this composition on service/docker level instead of root docker level ? Just curious if it can be confusing to peoples. @drasko @anovakovic01 ?

File: writers/influxdb/README.md [hunk 1]
```diff
@@ -42,6 +42,13 @@ Pass list of env variables in form `VARIABLE_NAME=[value]` separated by space ch
 Env variables are provided in table above. For example:
 MF_NATS_URL=nats://localhost:456 MF_DB_USER=user $GOBIN/mainflux-influxdb
 
+### Using docker-compose
+
+This service can also be deployed using docker containers.
+Docker compose file is available in ./docker/docker-compose.yml. Besides database
```
Comment (anovakovic01) [inline writers/influxdb/README.md]: Could you please change this path to be relative to project root?

File: writers/influxdb/docker/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    container_name: mainflux-influxdb-writer
+    depends_on:
+      - influxdb
+      - nats
+    expose:
+      - 8900
+    restart: on-failure
+    environment:
+      MF_NATS_URL: nats://nats:4222
+      MF_INFLUXDB_WRITER_PORT: 8900
```
Comment (anovakovic01) [inline writers/influxdb/docker/docker-compose.yml]: Don't forget to remove DB from env variable name.

File: writers/influxdb/docker/docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+      MF_INFLUXDB_DB_NAME: mainflux
+      MF_INFLUXDB_DB_HOST: influxdb
+      MF_INFLUXDB_DB_PORT: 8086
+      MF_INFLUXDB_DB_USER: mainflux
+      MF_INFLUXDB_DB_PASS: mainflux
+    ports:
+      - 8900:8900
+
+  grafana:
+    image: grafana/grafana:master
```
Comment (anovakovic01) [inline writers/influxdb/docker/docker-compose.yml]: Could you specify Grafana version?

**Review by anovakovic01 — CHANGES_REQUESTED** [review]
@nmarcetic I don't see any problem with docker-compose path, as it can be treated as addon.

**Comment by dborovcanin** [comment]
@nmarcetic This compose file is really about InfluxDB and its integration with Grafana, so it made sense for me to put it in service docker level.

**Comment by drasko** [comment]
@nmarcetic we have one generic Dockerfile that builds docker images for all services depending on parameters passed by Makefile: https://github.com/mainflux/mainflux/blob/master/docker/Dockerfile

I think this one should be in a separated, more generic, dir, as it is now.

`docker-compose` also uses external configs (`nginx.conf`) and certs, so this one also better live in separate dir, not with every service.

**Comment by drasko** [comment]
@dusanb94 nope, please move it to `docker` dir and call it `docker-compose-influxdb-grafana.yml`

**Comment by anovakovic01** [comment]
@drasko Let him at least place it in `docker/influxdb-grafana/docker-compose.yml`. We don't need complex docker-compose file name. Also, Makefile doesn't use only one generic Dockerfile. It uses Dockerfiles from `mqtt` and `dashflux`.

File: writers/influxdb/docker/docker-compose.yml (hunk 3 above)
Comment (nmarcetic) [reply writers/influxdb/docker/docker-compose.yml]: Agree @dusanb94 Please use latest.

**Comment by nmarcetic** [comment]
@anovakovic01 @dusanb94 I understand now and YES I agree it makes sense more to keep it on service level.

**Review by anovakovic01 — DISMISSED** [review]

**Review by nmarcetic — DISMISSED** [review]
LGTM!

**Review by drasko — DISMISSED** [review]
LGTM

**Comment by drasko** [comment]
@dusanb94 please update branch so we can merge it.

File: docker/addons/influxdb/docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    depends_on:
+      - influxdb
+      - nats
+    expose:
+      - 8900
+    restart: on-failure
+    environment:
+      MF_NATS_URL: nats://nats:4222
+      MF_INFLUXDB_WRITER_PORT: 8900
+      MF_INFLUXDB_POINT: messages
```
Comment (anovakovic01) [inline docker/addons/influxdb/docker-compose.yml]: Update this env var name.

**Review by anovakovic01 — CHANGES_REQUESTED** [review]

**Review by anovakovic01 — APPROVED** [review]
LGTM!

**Review by nmarcetic — APPROVED** [review]

**Pull request merged**